packages/next/lib/load-custom-routes.ts

```diff
@@ -668,6 +668,23 @@ export default async function loadCustomRoutes(
     }
   }
 
+  const totalRewrites =
+    rewrites.beforeFiles.length +
+    rewrites.afterFiles.length +
+    rewrites.fallback.length
+
+  const totalRoutes = headers.length + redirects.length + totalRewrites
```

Comment:
I also think that `trailingSlash` and `cleanUrls` will each add a route, right?

Reply:
The `trailingSlash` one is added above so should be included in this count, we could set the limit for the warning to an even 1000 to allow wiggle room

Reply:
Updated the limit and added the count before we add the `trailingSlash` redirects since including these in the count might cause confusion since they aren't directly adding them

Reply:
SGTM! Just need to update the PR desc